AutoStep/pPump.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+* This program is contains functions for the pPump library
+* Programmers: CJD
+* Editor: 
+* Iteration: 0.2
+* Last edited: 02/03/2021
+* Created: 10/08/2021
+* 
+* Dependencies: pPump.h, Arduino.h
+*
+* Library contrains fuctions for driving peristaltic pumps as they will be used for Boozebot
+* (Will be expanded to include linear rail functions and protothreading in next iteration)
+*/
+
+#include "Arduino.h"
+#include "pPump.h"
+
+pPump::pPump(int stepPin, int dirPin, int buttonPin, int period)
+{
+    pinMode(stepPin, OUTPUT);
+    pinMode(dirPin, OUTPUT);
+    pinMode(buttonPin, INPUT);
+    _stepPin = stepPin;
+    _dirPin = dirPin;
+    _buttonPin = buttonPin;
+    _period = period;
+}
+
+/* subfunction used by pour()
+    function accelerates stepper motor to its final period
+    important to avoid motor stalling */
+void accelerationStepper(int initial, int last, int increment, int stepPin) {
```

Comment:
Function starts acceleration process which is called once to avoid static stalling.